superset-frontend/src/dashboard/components/SliceHeader/index.tsx

```diff
@@ -176,6 +183,17 @@ const SliceHeader = forwardRef<HTMLDivElement, SliceHeaderProps>(
       ({ dashboardInfo }) => dashboardInfo.crossFiltersEnabled,
     );
 
+    const chart = useSelector<RootState, Chart>(
+      state => state.charts[slice.slice_id],
+    );
+
+    const theme = useTheme();
+
+    const firstQueryResponse = chart.queriesResponse?.[0];
+
+    const rowLimit = Number(formData.row_limit || 0);
+    const sqlRowCount = Number(firstQueryResponse?.sql_rowcount || 0);
```

Comment:
Is falling back to 0 safe here? If no row limit is set and there are no results from the query, wouldn't this show the warning incorrectly?